scripts/test/lib/cluster.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+function load_cluster_details() {
+  CLUSTER_INFO=$(aws eks describe-cluster --name $CLUSTER_NAME --region $REGION $ENDPOINT_FLAG)
+  VPC_ID=$(echo $CLUSTER_INFO | jq -r '.cluster.resourcesVpcConfig.vpcId')
+  SERVICE_ROLE_ARN=$(echo $CLUSTER_INFO | jq -r '.cluster.roleArn')
+  K8S_VERSION=$(echo $CLUSTER_INFO | jq -r '.cluster.version')
+  ROLE_NAME=${SERVICE_ROLE_ARN##*/}
+
+  echo "VPC ID: $VPC_ID, Service Role ARN: $SERVICE_ROLE_ARN, Role Name: $ROLE_NAME"
+}
+
+# This operation fails with rate limit exceeded when test is running for multiple K8s
+# version at same time, hence we increase the exponential retries on the aws call
+function attach_controller_policy_cluster_role() {
+  echo "Attaching IAM Policy to Cluster Service Role"
+  AWS_MAX_ATTEMPTS=10 aws iam attach-role-policy \
+    --policy-arn arn:aws:iam::aws:policy/AmazonEKSVPCResourceController \
+    --role-name "$ROLE_NAME" > /dev/null
+}
+
+function detach_controller_policy_cluster_role() {
+  echo "Detaching the IAM Policy from Cluster Service Role"
+  aws iam detach-role-policy \
+    --policy-arn arn:aws:iam::aws:policy/AmazonEKSVPCResourceController \
+    --role-name $ROLE_NAME > /dev/null
```

Comment:
1. What happens if there is a failure in Deletion, does the script terminate ? 
2. Wondering if we should preserve the output of these commands instead > /dev/null, would help us debug if there are any errors.

Reply:
1. Yeah, the script would exit as we have "set -e"
2. I think the error would still be printed out, did a small test by trying to detach the policy from a role which doesn't have the policy:
`An error occurred (NoSuchEntity) when calling the DetachRolePolicy operation: Policy arn:aws:iam::aws:policy/AmazonEKSVPCResourceController was not found.`